### DISCUSSION TITLE
Internal Server Error when doing anything

### DISCUSSION THREAD

**Discussion by vertekal1** [sheltermanager/asm3#1545] [Q&A / Help] [answered]
I recently did a fresh install of ASM3 (https://public.sheltermanager.com/deb/sheltermanager3_44_all.deb) on an Ubuntu 22.04 virtual machine (hosted on Azure). I believe I configured pg_hba.conf and asm3.conf correctly, and can access my ASM site and log in. But, doing anything (adding an animal, for example) results in an "Internal Server Error" at the top of the screen. I checked the Apache2 log, and the error appears to be AttributeError: 'HTMLParser' object has no attribute 'unescape'

A bit of googling tells me that HTMLParser is no longer valid in the current version of Python.

Is there a way I can fix this?

**Comment by vertekal1** [comment]
I updated /usr/lib/sheltermanager3/asm3/utils.py and added a line for "import html" at the top, and then changed the line 

parser = HTMLParser()
return parser.unescape(s)

to

#parser = HTMLParser()
return html.unescape()

This allowed me to add an animal. I'm guessing I'll have to make a similar update to other py files.

**Comment by bobintetley** [comment, marked as answer]
44 is pretty old, is there a reason you didn't go with 48?

https://sheltermanager.com/site/en_oss.html

**Reply by vertekal1** [reply]
I'm not terribly familiar with Linux and how to install things. I wasn't sure how to get the .deb file to my server, and found the link to 44 on a random website (https://lesbianunix.dev/posts/asm3/) that came up when I was looking for any type of installation notes (aside from the readme file). 

Can I install 48 on top of 44?

**Comment by vertekal1** [comment]
I ended up just spinning up a new Ubuntu VM and installed 48. Everything seems to be working properly. Thanks!
